libraries/tlv-account-resolution/src/key_data.rs

```diff
@@ -0,0 +1,183 @@
+//! Types for managing extra account meta keys that may be extracted from some
+//! data.
+//!
+//! This can be either account data from some account in the list of accounts
+//! or from the instruction data itself.
+
+#[cfg(feature = "serde-traits")]
+use serde::{Deserialize, Serialize};
+use {crate::error::AccountResolutionError, solana_program::program_error::ProgramError};
+
+/// Enum to describe a required key stored in some data.
+#[derive(Clone, Debug, PartialEq)]
+#[cfg_attr(feature = "serde-traits", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde-traits", serde(rename_all = "camelCase"))]
+pub enum KeyData {
```

Comment:
nit: Since we use `new_with_pubkey` elsewhere for the raw pubkey data, how about renaming all instances of `KeyData` to `PubkeyData`? I find the term "key" to be a little less clear

Reply:
Sure, I like that better!